pkg/controller/job/job_controller.go

```diff
@@ -1041,8 +1042,11 @@ func (jm *Controller) trackJobStatusAndRemoveFinalizers(ctx context.Context, job
 			break
 		}
 	}
-	if len(newSucceededIndexes) > 0 {
+	if isIndexed {
 		succeededIndexes = succeededIndexes.withOrderedIndexes(newSucceededIndexes)
```

Comment:
I think, at this point `succeededIndexes-prevSucceededIndexes` should have the delta.

Reply:
Did you try this? `succeededIndexes.total() - prevSucceededIndexes.total()`

Reply:
Yes, it results in the test case indexed job past and new pods failing due to `recordJobPodFinished()` being called twice during the test, so simply passing in this delta results in double counting.